### PR TITLE
Unknown identifier fixes (Lean).

### PR DESCRIPTION
**Pull request by MichaelHenryJennings**

### CODE REVIEW

**Comment by eric-wieser** [comment]
Nice, this closes #202 I think

File: lean4/src/putnam_2017_b2.lean [hunk 1]
```diff
@@ -11,7 +11,7 @@ theorem putnam_2017_b2
 (S : ℤ → ℕ → ℤ)
 (p : ℤ → ℕ → Prop)
 (q : ℤ → Prop)
-(hS : S = fun a k ↦ ∑ i : Fin k, a + i)
+(hS : S = fun (a : ℤ) k ↦ ∑ i in Finset.range k, (a + i))
```
Comment (eric-wieser) [inline lean4/src/putnam_2017_b2.lean:14]: You shouldn't need the type ascription here

Comment (MichaelHenryJennings) [reply lean4/src/putnam_2017_b2.lean:14]: When I remove it, I get the error:
> type mismatch
  fun a k => ∑ i ∈ Finset.range k, (a + i)
has type
  ℕ → ℕ → ℕ : Type
but is expected to have type
  ℤ → ℕ → ℤ : Type

Comment (eric-wieser) [reply lean4/src/putnam_2017_b2.lean:14]: Huh, strange; that wasn't what I expected!

**Review by eric-wieser — APPROVED** [review]
Thanks!

**Review by amit9oct — APPROVED** [review]

**Pull request merged**